prow/cmd/webhook-server/helpers.go

```diff
@@ -131,3 +185,151 @@ func isCertValid(cert string) error {
 	}
 	return nil
 }
+
+func getGCPSecrets(client ClientInterface, ctx context.Context, expiry int, dns prowflagutil.Strings) (string, string, string, error) {
+	secretsMap := make(map[string]string)
+	data, err := client.GetSecretValue(ctx, secretID, "latest")
+	if err != nil {
+		return "", "", "", fmt.Errorf("%s, unable to get secret value: %v", gcpSecretError, err)
+	}
+
+	err = json.Unmarshal(data, &secretsMap)
+	if err != nil {
+		return "", "", "", fmt.Errorf("error marshalling CA cert secret data: %v", err)
+	}
+
+	cert := secretsMap[caCert]
+	privKey := secretsMap[caPrivKey]
+	caPem := secretsMap[caBundle]
+
+	return cert, privKey, caPem, nil
+}
+
+
+func createGCPSecret(client ClientInterface, ctx context.Context, expiry int, dns []string) (string, string, string, error) {
+	if _, err := client.CreateSecret(ctx, secretID); err != nil {
+		return "", "", "", fmt.Errorf("unable to create secret %v", err)
+	}
+
+	serverCertPerm, serverPrivKey, caPem,  err := updateGCPSecret(client, ctx, expiry, dns)
+	if err != nil {
+		return "", "", "", fmt.Errorf("unable to write secret value %v", err)
+	}
+	return serverCertPerm, serverPrivKey, caPem, nil
+}
+
+func updateGCPSecret(client ClientInterface, ctx context.Context, expiry int, dns[] string) (string, string, string, error) {
+	serverCertPerm, serverPrivKey, caPem, secretData, err := genSecretData(expiry, dns)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	if err := client.AddSecretVersion(ctx, secretID, secretData); err != nil {
+		return "", "", "", fmt.Errorf("unable to add secret version %v", err)
+	}
+
+	return serverCertPerm, serverPrivKey, caPem, nil
+}
+
+func genSecretData(expiry int, dns []string) (string, string, string, []byte, error) {
+	serverCertPerm, serverPrivKey, caPem, err := genCert(expiry, dns)
+	if err != nil {
+		return "", "", "",  nil, fmt.Errorf("could not generate ca credentials")
+	}
+	caSecrets := map[string]string{
+	    caCert: serverCertPerm,
+	    caPrivKey: serverPrivKey,
+	    caBundle: caPem,
+	}
+	secretData, err := json.Marshal(caSecrets)
+
+	if err != nil {
+		return "", "", "", nil, fmt.Errorf("error unmarshalling CA cert secret data: %v", err)
+	}
+
+	return serverCertPerm, serverPrivKey, caPem, secretData, nil
+}
+
+func createValidatingWebhookConfig(ctx context.Context, caPem string, client ctrlruntimeclient.Client) error {
```

Comment:
this change is not part of FileSystem change, is this PR based off of another PR?

Reply:
decided to sneak in config changes here since its a small but necessary addition

Reply:
ill update the description of the pr